src/index.ts

```diff
@@ -146,6 +165,33 @@ export class EthrDID {
     return receipt.transactionHash
   }
 
+  async createAddDelegateHash(delegateType: string, delegateAddress: address, exp: number): Promise<string> {
+    if (typeof this.controller === 'undefined') {
+      throw new Error('a web3 provider configuration is needed for network operations')
+    }
+    return this.controller.createAddDelegateHash(delegateType, delegateAddress, exp)
+  }
+
+  async addDelegateSigned(
+    delegate: string,
+    signature: MetaSignature,
+    delegateOptions?: DelegateOptions,
+    txOptions: CallOverrides = {}
+  ): Promise<string> {
+    if (typeof this.controller === 'undefined') {
+      throw new Error('a web3 provider configuration is needed for network operations')
+    }
+    const owner = await this.lookupOwner()
+    const receipt = await this.controller.addDelegateSigned(
+      delegateOptions?.delegateType || DelegateTypes.veriKey,
+      delegate,
+      delegateOptions?.expiresIn || 86400,
+      signature,
+      { ...txOptions, from: owner }
+    )
```

Comment:
The `*Signed()` methods don't require the sender to be the current owner.

```suggestion
    const receipt = await this.controller.addDelegateSigned(
      delegateOptions?.delegateType || DelegateTypes.veriKey,
      delegate,
      delegateOptions?.expiresIn || 86400,
      signature,
      txOptions
    )
```